plugin/sublime-text/plugin_subl3.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (C) 2020  Matthew "strager" Glazar
+# See end of file for extended copyright information.
+
+"""Shows to the user the diagnostics provided by quick-lint-js."""
+
+# Just for the sake of clarity, you can think of a buffer as a block of
+# memory that contains the file's text and a view as a tab in the
+# sublime text.
+
+import sublime
+import sublime_plugin
+
+from . import c_api
+
+
+class Buffer:
+    """Represents the file's text buffer.
+
+    Multiple view objects may share the same buffer.
+    """
+
+    def __init__(self, view):
+        self.views = [view]
+        self.parser = c_api.Parser(view)
+
+
+class BuffersManager:
+    """Manages the buffers."""
+
+    def __init__(self):
+        self.buffers = {}
+
+    def add_view(self, view):
+        id_ = view.buffer_id()
+        if id_ not in self.buffers:
+            self.buffers[id_] = Buffer(view)
+        else:
+            self.buffers[id_].views.append(view)
+
+    def remove_view(self, view):
+        id_ = view.buffer_id()
+        self.buffers[id_].views.remove(view)
+        if not self.buffers[id_].views:
+            del self.buffers[id_]
+
+    def get_buffer(self, view):
+        id_ = view.buffer_id()
+        return self.buffers[id_]
+
+
+class QuickLintJsListener(sublime_plugin.ViewEventListener):
+    """Listens for events bound to a specific view."""
+
+    # The internal strategy used is to share information between
+    # all views that belong to the same buffer. Because that way,
+    # if there are multiple views (tabs) of the same buffer (file),
+    # they will all apply the same changes (have squiggly underlines
+    # and pop-ups available):
+    #
+    # https://github.com/quick-lint/quick-lint-js/pull/328#issuecomment-869038036
+
+    buffers_manager = BuffersManager()
+
+    @classmethod
+    def is_applicable(cls, settings):
+        syntax = settings.get("syntax", "")
+        if "JavaScript" in syntax:
+            return True
+        return False
+
+    @classmethod
+    def applies_to_primary_view_only(cls):
+        return False
+
+    def __init__(self, view):
+        super().__init__(view)
+        self.buffers_manager.add_view(self.view)
+        self.buffer = self.buffers_manager.get_buffer(self.view)
+        self.on_modified()
```

Comment:
> Won't `on_load` get called anyway? Why do we need to explicitly lint in `__init__`?


Sublime Text calls the `__init__` method every time the tab (view) is loaded.

Sublime Text calls the `on_load` method every time the file (buffer) is loaded.

We need to explicitly lint in `__init__` because there are some situations where Sublime Text has already loaded the file, and, in these situations, Sublime Text calls `__init__` and on_load doesn't:

```
The `__init__` method was called.
Timestamp: 1626315449.1726804
File content: const 🎸 = "asdf";
```

We need to explicitly lint in `on_load` because there are some situations where Sublime Text needs to load the file, and, in these situations, Sublime Text calls `__init__` and `on_load` too, when `__init__` runs lint, lint gets an empty text, but when `on_load` runs lint, lint receives the entire file content:

```
The `__init__` method was called.
Timestamp: 1626315480.3791265
File content: 

The `on_load` method was called.
Timestamp: 1626315480.4387884
File content: const 🎸 = "asdf";
```